spec/lib/service/runner_spec.rb

```diff
@@ -167,6 +167,26 @@ def non_raising_step
     end
   end
 
+  class LockService
+    include Service::Base
+
+    params do
+      attribute :post_id, :integer
+      attribute :user_id, :integer
+    end
+
+    lock(:post_id, :user_id) { step :locked_step }
+    step :after_lock
+
+    def locked_step
+      @ran = true
+    end
+
+    def after_lock
```

Comment:
This step is not needed actually, since the `lock` step will fail and stop the entire service if the lock isn’t acquired. `locked_step` can also just be a no-op (or just returns `true`).

Reply:
We should really remove the `after_lock` step as it’s pretty useless.